### PR TITLE
TST: Suppress warnings of drop_duplicates tests

### PR DESCRIPTION
**Pull request by sinhrks**
Follow-up of #10236. Suppress unnecessary warnings during the test.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
I don't think this is the correct fix. You shouldn't need to specify the `warnings.simplefilter('always', FutureWarning)` everywhere. This is done by `assert_produces_warning`. Or am I missing something?

**Comment by sinhrks** [comment]
Correct, fixed to use ``clear`` argument of ``assert_produces_warning``.

File: pandas/tests/test_base.py [hunk 1]
```diff
@@ -730,11 +726,11 @@ def test_duplicated_drop_duplicates(self):
                 tm.assert_index_equal(result, idx[~expected])
 
                 # deprecate take_last
-                with tm.assert_produces_warning(FutureWarning):
+                with tm.assert_produces_warning(FutureWarning, clear=pd.core.base):
```
Comment (jreback) [inline pandas/tests/test_base.py]: we don't normally pass ``clear``. seems this should in fact default to the current module (but can leave that for another PR).

**Comment by jorisvandenbossche** [comment]
@sinhrks do you know why the `clear` is needed in this case (otherwise we almost never use it)?

**Comment by jreback** [comment]
lgtm, merge when ready.

**Comment by sinhrks** [comment]
Actually ``clear`` is not needed if we use ``assert_produces_warning`` in all cases. 

@jorisvandenbossche Pls merge if looks OK.

**Comment by jorisvandenbossche** [comment]
ah yes! Merging, thanks!
